compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs

```diff
@@ -129,8 +133,43 @@ newtype IgnorePackageMetadata = IgnorePackageMetadata { getIgnorePackageMetadata
 newtype Haddock = Haddock Bool
   deriving Show
 
+-- | The dlint rules file is a dlint yaml file that's used as the base for
+-- the rules used during linting. Really there is no difference between the
+-- rules file and the other hint files, but it is useful to specify them
+-- separately since this one can act as the base, allowing the other hint files
+-- to selectively ignore individual rules.
+data DlintRulesFile
+  = DefaultDlintRulesFile
+    -- ^ "WORKSPACE/compiler/damlc/daml-ide-core/dlint.yaml"
+  | ExplicitDlintRulesFile FilePath
+    -- ^ User-provided rules file
+  deriving Show
+
+data DlintHintFiles
+  = ImplicitDlintHintFile
+    -- ^ First existing file of
+    --    *       ".dlint.yaml"
+    --    *    "../.dlint.yaml"
+    --    * "../../.dlint.yaml"
+    --    * ...
+    --    * "~/.dlint.yaml"
+  | ExplicitDlintHintFiles [FilePath]
+  deriving Show
+
+data DlintOptions = DlintOptions
+  { dlintRulesFile :: DlintRulesFile
+  , dlintHintFiles :: DlintHintFiles
+  }
+  deriving Show
+
+defaultDlintOptions :: DlintOptions
+defaultDlintOptions = DlintOptions
+  { dlintRulesFile = DefaultDlintRulesFile
+  , dlintHintFiles = ExplicitDlintHintFiles []
+  }
+
 data DlintUsage
-  = DlintEnabled { dlintUseDataDir :: FilePath, dlintAllowOverrides :: Bool }
+  = DlintEnabled DlintOptions
```

Comment:
Now that the record has its own type, it's easier to write functions where we only care about the `Enabled` case.